### PR TITLE
Merge More Main Changes to Azure Storage v12 Branch

### PR DESCRIPTION
**Pull request by wsugarman**
- Integrate latest changes from `main` including the new table partition manager into the feature branch
- Refactor `TablePartitionManager` to use the `Table` class instead of the `TableClient` directly
- Rename `TableLease` to `TablePartitionLease` to align with `BlobPartitionLease`
- Some async testing refactors

This change should **not** be squashed as I will eventually need to merge with `main` too.

### CODE REVIEW

**Comment by cgillum** [comment]
Adding @nytian to help review this PR and answer Will’s questions.

File: src/DurableTask.AzureStorage/Storage/Table.cs [hunk 1]
```diff
@@ -69,37 +69,42 @@ public async Task<bool> ExistsAsync(CancellationToken cancellationToken = defaul
             return tables.Count > 0;
         }
 
-        public async Task ReplaceAsync<T>(T tableEntity, ETag ifMatch, CancellationToken cancellationToken = default) where T : ITableEntity
+        public async Task DeleteAsync(CancellationToken cancellationToken = default)
+        {
+            await this.tableClient.DeleteAsync(cancellationToken).DecorateFailure();
```
[the remark this answers is not in the document]
Comment (nytian) [reply src/DurableTask.AzureStorage/Storage/Table.cs:74]: That's good question I didn't aware of. Let me ask function host team if we can rely on this new package or they have plan to upgrade the version.

**Comment by nytian** [comment]
The change looks good to me. I just have one problem, why are we changing the TableClient to Table, I assume is it because we cannot apply Azure.Data.Tables packages? Thanks!

**Review by nytian — APPROVED** [review]
The change looks good to me. 
I just have a problem about changing TableClient to Table class.  Why are we doing this, I assume is it because we cannot apply Azure.Data.Tables packages? 

And I also tried to use the Table class in my implementation before, but it will always remind me this error. I am just curious what I missed. :)  
image
Thanks!

**Comment by wsugarman** [comment]
> The change looks good to me. I just have a problem about changing TableClient to Table class. Why are we doing this, I assume is it because we cannot apply Azure.Data.Tables packages? Thanks!

Oh, I changed it as all other interactions with Azure Storage, including the existing partition manager, leverage these client wrappers. For example, the `BlobPartitionManager` used by the `LegacyPartitionManager` [uses the internal `Blob` type](https://github.com/Azure/durabletask/blob/1bb5c8f964f085bcdd6268c0fb5ec76a0701fb5a/src/DurableTask.AzureStorage/Partitioning/BlobLeaseManager.cs#L307). Probably the biggest advantage is that they are configured with [various policies for retrying and logging connection attempts](https://github.com/Azure/durabletask/blob/cd477c749c71fd4fd16aaf3a245c0c145c5ac975/src/DurableTask.AzureStorage/Storage/AzureStorageClient.cs#L61).

**Pull request merged**